src/acl/Gadgets.cc

```diff
@@ -184,64 +191,25 @@ aclParseAclList(ConfigParser &, Acl::Tree **treep, const char *label)
     rule->context(ToSBuf('(', cfg_directive, ' ', label, " line)"), config_input_line);
     const auto aclCount = rule->lineParse();
 
-    // We want a cbdata-protected Tree (despite giving it only one child node).
+    // XXX: We have created only one node, and our callers do not support
+    // actions, but we now have to create an action-supporting Acl::Tree because
+    // Checklist needs actions support. TODO: Add actions methods to Acl::Node,
+    // so that Checklist can be satisfied with Acl::AndNode created above.
```

Comment:
The problem described by this XXX was not added by this PR. The earlier TODO statement was incomplete -- this code needed not just cbdata protection but actions support. We have discovered this other need when trying to address that TODO in this PR (after removing the need for cbdata protection). This problem is now documented (as an XXX rather than just TODO). Fixing it is not difficult but should be done in a dedicated PR because the problem and the fix are not related to reference counting.